packages/@guardian/source-foundations/src/typography/font-styles.ts

```diff
@@ -128,12 +133,15 @@ export const fontStyleToStringFunction =
 			${fontWeight ? `font-weight: ${fontWeight}` : ''};
 			${fontStyle ? `font-style: ${fontStyle}` : ''};
 
-			&:hover {
-				${
-					textDecorationThickness
-						? `text-decoration-thickness: ${textDecorationThickness}px`
-						: ``
-				};
-			}
+			/*
+			 * Child elements (e.g. <Link/>) can use this variable
+			 * to set the thickness of their underline on hover.
+			 *
+			 * The thickness for each font type and weight is defined
+			 * in the underlineThickness object.
+			 */
+			--source-hover-text-decoration-thickness: ${
+				textDecorationThickness ? `${textDecorationThickness}px` : 'auto'
```

Comment:
Will people ever want to set `textDecorationThickness` to zero? And would this come through as a `Number`? If so, best to do `textDecorationThickness == null ? ...` perhaps to avoid falsey zeros.

Reply:
good spot — right now the text decoration thickness is pulled from [this mapping](https://github.com/guardian/source/blob/main/packages/%40guardian/source-foundations/src/typography/data.ts#L118) so we can customise it for different font/weight combinations.

that mapping doesn't return 0 right now - but could in the future, so I think specifically checking for `undefined` here makes sense 👍